### PR TITLE
wxString move/rvalue backport

### PR DESCRIPTION
**Pull request by pavel-t**

### CODE REVIEW

**Review by vadz — COMMENTED** [review]
Thanks, do you plan any more changes here or should we merge this?

I _think_ it's ABI-compatible and so we could have it in 3.2.2, but maybe it would be safer to leave it until after it... not completely sure.

File: include/wx/string.h [hunk 1]
```diff
@@ -1140,12 +1140,25 @@ class WXDLLIMPEXP_BASE wxString
...
   static wxString FromImpl(const wxStringImpl& src)
       { return wxString((CtorFromStringImplTag*)NULL, src); }
+
+  #ifdef wxHAS_RVALUE_REF
+  wxString(CtorFromStringImplTag* WXUNUSED(dummy), wxStringImpl&& src) wxNOEXCEPT
+      : m_impl(std::move(src)) {}
+
+  static wxString FromImpl(wxStringImpl&& src) wxNOEXCEPT
+  {
+      return wxString((CtorFromStringImplTag*)nullptr, std::move(src));
```
[the remark this answers is not in the document]
Comment (pavel-t) [reply include/wx/string.h:1150]: Changing to NULL for consistency with lvalue version

**Comment by vadz** [comment]
Sorry, I didn't have time to recheck/retest this before 3.2.2, but I'll merge this once it is done and it will be part of 3.2.3 if no problems are found.

**Comment by vadz** [comment]
Thanks again, (squash) merged to 3.2 branch now.

**Pull request closed without merging**